### PR TITLE
docs: improve self-hosted docs

### PR DESCRIPTION
**Pull request by owlas**
Closes: #4822 
Closes: #4942 

### Description:

- New guide for a minimum self-hosted setup
- Pros/cons of self-hosting
- List prerequisites for self hosting
- Improve docs for configuring self-hosted options (schedulers, email, slack, object storage, ...)
- Centralised all docs related to self-hosted Lightdash to the self-host section

### CODE REVIEW

File: docs/docs/self-host/secure-lightdash-with-https.mdx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Secure Lightdash with HTTPS
+
+To run Lightdash in production it is highly recommended to secure it with HTTPS. This is usually done using your
```
Comment (rephus) [inline docs/docs/self-host/secure-lightdash-with-https.mdx:3]: Fun fact, you can't connect Slack if you don't have an HTTPS callback url (there is a workaround though, that I use on localhost)

File: docs/docs/self-host/self-host-lightdash-docker-compose.mdx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- `LIGHTDASH_SECRET` is the secret used to encrypt data at rest in the database. If you lose this secret, you will not be able to access your data in Lightdash.
+
+```bash
+
+```bash
+LIGHTDASH_SECRET="not very secret" PGPASSWORD="password" \
+  docker-compose -f docker-compose.yml up --detach --remove-orphans
+```
+
+:::info
```
Comment (rephus) [inline docs/docs/self-host/self-host-lightdash-docker-compose.mdx:31]: There is another very common error for users on Linux not running the daemon... I think we even have an issue about it..

File: docs/docs/self-host/self-host-lightdash-restack.mdx [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# Self-Host Lightdash with Restack
+
+Restack is a managed service for deploying and managing open-source software.
+
+## Deploy to AWS with Restack
```
Comment (rephus) [inline docs/docs/self-host/self-host-lightdash-restack.mdx:5]: cool, have you tried this ?

File: docs/docs/self-host/self-host-lightdash-restack.mdx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+5. Enter your *AWS Access Key ID* and *AWS Secret Access key*.
+6. Click *Add credential*.
+   ![add-aws-credentials](../assets/restack-add-AWS-creds.png)
+
+>[How to get your AWS Access key ID and AWS Secret Access Key](https://docs.aws.amazon.com/accounts/latest/reference/root-user-access-key.html)
+
+### One-click cluster creation with Restack
+
+:::tip
+Why do I need a cluster?<br/>
```
Comment (rephus) [inline docs/docs/self-host/self-host-lightdash-restack.mdx:42]: What's are all these <br/> ? :suspect: did chatgpt write this for you ?

File: docs/docs/self-host/self-host-lightdash.mdx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+This guide is designed for DevOps Engineers that are familiar with Docker, Kubernetes, and are comfortable 
+configuring environment variables, SMTP credentials, and database connections.
+If you're unsure whether to self-host please read our guide on [Lightdash Cloud vs. Self-Hosted](./lightdash-cloud-vs-self-hosted.mdx).
+
+## Prerequisites for self-hosting
+
+ - Access to a kubernetes cluster and [`kubectl`](https://kubernetes.io/docs/tasks/tools/) installed
+ - [Helm](https://helm.sh/docs/intro/quickstart/)
+ - Docker
```
Comment (rephus) [inline docs/docs/self-host/self-host-lightdash.mdx:13]: Also `sops`

File: docs/docs/self-host/self-host-lightdash-restack.mdx (hunk 4 above)
Comment (owlas) [reply docs/docs/self-host/self-host-lightdash-restack.mdx:42]: This is already in the docs! This was contributed by the team at Restack, I believe it's a template set of docs they use. I'm happy to keep it in

File: docs/docs/self-host/self-host-lightdash.mdx (hunk 5 above)
Comment (owlas) [reply docs/docs/self-host/self-host-lightdash.mdx:13]: Only if using `helm secrets` but I don't mention that in the docs here, I'll leave secrets management to the user.

File: docs/docs/self-host/self-host-lightdash-restack.mdx (hunk 3 above)
Comment (owlas) [reply docs/docs/self-host/self-host-lightdash-restack.mdx:5]: Nope! This was already in the docs. We have some users on restock

File: docs/docs/self-host/self-host-lightdash-docker-compose.mdx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+# Clone the Lightdash repo
+git clone https://github.com/lightdash/lightdash
+cd lightdash
+```
+
+## 2. Create containers
+
+You must set the following two environment variables:
+
+- `PGPASSWORD` is the password used for the internal postgres database
```
[the remark this answers is not in the document]
Comment (owlas) [reply docs/docs/self-host/self-host-lightdash-docker-compose.mdx:21]: I can do this, I listed them in the code snippet below but I didn't export them!

File: docs/docs/self-host/self-host-lightdash-docker-compose.mdx [hunk 7]
```diff
@@ -0,0 +1,36 @@
+# Self-host Lightdash using docker-compose
+
+This guide will give you a minimal Lightdash instance running on your local machine. It will not be accessible from 
+the internet, but it will be accessible from your local machine. This is a great way to get started with Lightdash 
+for a proof-of-concept without needing access to kubernetes.
+
+## 1. Clone the Lightdash repository
```
[the remark this answers is not in the document]
Comment (owlas) [reply docs/docs/self-host/self-host-lightdash-docker-compose.mdx:7]: Good point!

File: docs/docs/self-host/secure-lightdash-with-https.mdx (hunk 1 above)
Comment (owlas) [reply docs/docs/self-host/secure-lightdash-with-https.mdx:3]: This makes sense, also for SSO. I'll mention that on the docs for those!

File: docs/docs/self-host/lightdash-cloud-vs-self-hosted.mdx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+sidebar_position: 2
+sidebar_label: Cloud vs. Self-Hosted
+---
+
+# Lightdash Cloud vs. Self-Hosted
+
+Lightdash is open-source and licensed under the MIT license. This means that you are free to self-host Lightdash on 
+your own infrastructure but without any warranty or support from the Lightdash team. In other words, it's up to you 
+to maintain and monitor your Lightdash instance.
+
```
Comment (TuringLovesDeathMetal) [inline docs/docs/self-host/lightdash-cloud-vs-self-hosted.mdx:11]: This is a great addition! 

I wonder if we can be even more like "harsh" here? As in, being quite explicit about needing to basically be a data engineer if you want to self-host Lightdash. You need to be comfortable with Docker, etc.?

**Review by TuringLovesDeathMetal — CHANGES_REQUESTED** [review]
Nice! This looks great. I like splitting out self-hosted into its own space (basically streamlining  docs to Cloud and pushing folks that way).

This was some of the stuff we wrote in the retro as ideas: 
image

**Comment by owlas** [comment]
Updated!

File: docs/docs/self-host/lightdash-cloud-vs-self-hosted.mdx (hunk 8 above)
Comment (owlas) [reply docs/docs/self-host/lightdash-cloud-vs-self-hosted.mdx:17]: Updated this! There is also a similar disclaimer at the top of the main self hosting page too

**Pull request merged**